src/topology/algebra/infinite_sum.lean

```diff
@@ -607,6 +607,17 @@ lemma set.finite.summable_compl_iff {s : set β} (hs : s.finite) :
   summable (f ∘ coe : sᶜ → α) ↔ summable f :=
 (hs.summable f).summable_compl_iff
 
+lemma has_sum_ite_eq_extract [decidable_eq β] (hf : has_sum f a) (b : β) :
+  has_sum (λ n, ite (n = b) 0 (f n)) (a - f b) :=
+begin
+  convert has_sum_subtype_iff_indicator.mp (({b} : finset β).has_sum_iff_compl.mp hf),
+  { ext,
+    rw ← ite_not,
+    congr,
+    simp },
+  { exact sum_singleton.symm }
+end
```

Comment:
It makes the proof significantly shorter:

```suggestion
begin
  convert hf.update b 0 using 1,
  { ext n, rw function.update_apply, },
  { rw [sub_add, zero_sub, neg_sub] },
end
```

Reply:
OK!